packages/circuit-ui/components/Dialog/DialogService.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright 2024, SumUp Ltd.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export const getKeyboardFocusableElements = (
+  element: HTMLElement,
+): HTMLElement[] =>
+  [
+    ...element.querySelectorAll(
+      'a[href], button, input, textarea, select, details,[tabindex]:not([tabindex="-1"])',
+    ),
+  ].filter(
+    (el) => !el.hasAttribute('disabled') && !el.getAttribute('aria-hidden'),
```

Comment:
Should we also filter for `aria-disabled` here, given that we use it in the Button component?